packages/loot-core/src/client/data-hooks/reports.ts

```diff
@@ -0,0 +1,44 @@
+import { useMemo } from 'react';
+
+import { q } from '../../shared/query';
+import {
+  type CustomReportData,
+  type CustomReportEntity,
+} from '../../types/models';
+import { useLiveQuery } from '../query-hooks';
+
+function toJS(rows: CustomReportData[]) {
+  const reports: CustomReportEntity[] = rows.map(row => {
+    const test: CustomReportEntity = {
+      ...row,
+      conditionsOp: row.conditions_op ?? 'and',
+      filters: row.conditions,
+    };
+    return test;
+  });
+  return reports;
+}
+
+/*
+leaving as a placeholder for saved reports implementation return an empty array because "reports" db table doesn't exist yet
+*/
+export function useReports(): CustomReportEntity[] {
+  const reports: CustomReportEntity[] = toJS(
+    //useLiveQuery(() => q('reports').select('*'), []) || [],
+    useLiveQuery(() => q('transaction_filters').select('*'), []) || [],
+  );
+
+  /** Sort reports by alphabetical order */
+  function sort(reports: CustomReportEntity[]) {
+    return reports.sort((a, b) =>
+      a.name
+        .trim()
+        .localeCompare(b.name.trim(), undefined, { ignorePunctuation: true }),
+    );
+  }
+
+  const flag = true;
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution.  -->
**❓ question:** ‏what's this?

Reply:
Tried to explain it in commnets in line 23.

"useReports" is meant to pull an array of all reports in the db table...except there's no db table yet. This is the work-around I implemented so that the code doesn't create fatal errors. It's basically returning an empty array but all this code will be used for querying the db for saved reports.